crates/astria-sequencer/src/app.rs

```diff
@@ -537,38 +519,72 @@ impl App {
         // however, we need to still construct a `BeginBlock` type for now as
         // the penumbra IBC implementation still requires it as a parameter.
         let begin_block = abci::request::BeginBlock {
-            hash: finalize_block.hash,
-            byzantine_validators: finalize_block.misbehavior.clone(),
+            hash: Hash::default(), // unused
+            byzantine_validators: block_data.misbehavior.clone(),
             header: Header {
                 app_hash: self.app_hash.clone(),
                 chain_id: chain_id.clone(),
-                consensus_hash: Hash::default(), // unused
-                data_hash: Some(Hash::try_from(data_hash.to_vec()).unwrap()),
+                consensus_hash: Hash::default(),      // unused
+                data_hash: Some(Hash::default()),     // unused
                 evidence_hash: Some(Hash::default()), // unused
-                height: finalize_block.height,
+                height: block_data.height,
                 last_block_id: None,                      // unused
                 last_commit_hash: Some(Hash::default()),  // unused
                 last_results_hash: Some(Hash::default()), // unused
-                next_validators_hash: finalize_block.next_validators_hash,
-                proposer_address: finalize_block.proposer_address,
-                time: finalize_block.time,
+                next_validators_hash: block_data.next_validators_hash,
+                proposer_address: block_data.proposer_address,
+                time: block_data.time,
                 validators_hash: Hash::default(), // unused
                 version: tendermint::block::header::Version {
                     // unused
                     app: 0,
                     block: 0,
                 },
             },
-            last_commit_info: finalize_block.decided_last_commit.clone(),
+            last_commit_info: tendermint::abci::types::CommitInfo {
+                round: 0u16.into(), // unused
+                votes: vec![],
+            }, // unused
         };
 
-        self.begin_block(&begin_block, storage.clone())
+        self.begin_block(&begin_block)
             .await
             .context("failed to call begin_block")?;
 
+        Ok(block_data.txs)
+    }
+
+    #[instrument(name = "App::finalize_block", skip_all)]
+    pub(crate) async fn finalize_block(
+        &mut self,
+        finalize_block: abci::request::FinalizeBlock,
+        storage: Storage,
+    ) -> anyhow::Result<abci::response::FinalizeBlock> {
+        let chain_id: tendermint::chain::Id = self
+            .state
+            .get_chain_id()
+            .await
+            .context("failed to get chain ID from state")?
+            .try_into()
+            .context("invalid chain ID")?;
+
+        // set the current proposer
+        self.current_proposer = Some(finalize_block.proposer_address);
```

Comment:
I think this is just used in `end_block()` which we call from this function. If there isn't a reason why don't want to change the `end_block()` signature, it might be cleaner to just pass it into that function and not set/unset it in this one.